cluster-autoscaler/cloudprovider/azure/azure_vms_pool.go

```diff
@@ -102,28 +120,260 @@ func (agentPool *VMsPool) MaxSize() int {
 // TargetSize returns the current TARGET size of the node group. It is possible that the
 // number is different from the number of nodes registered in Kubernetes.
 func (agentPool *VMsPool) TargetSize() (int, error) {
-	// TODO(wenxuan): Implement this method
-	return -1, cloudprovider.ErrNotImplemented
+	size, err := agentPool.getVMsPoolSize()
```

Comment:
why can't we just implement TargetSize directly rather than deferring to `getVMsPoolSize`? Can't we just use TargetSize in place of this method wherever we need it?

Reply:
I actually followed exact same pattern with VMSS here - thought that will make it easier for maintainers: https://github.com/kubernetes/autoscaler/blob/3914bdebd1e9453396408f1f99efa4cc5eeaf078/cluster-autoscaler/cloudprovider/azure/azure_scale_set.go#L334